src/api/Nhs.Appointments.Api/Functions/QueryAvailabilityFunction.cs

```diff
@@ -62,7 +63,14 @@ protected override async Task<ApiResult<QueryAvailabilityResponse>> HandleReques
         var requestUntil = request.Until;
         var requestConsecutive = request.Consecutive;
 
-        await Parallel.ForEachAsync(request.Sites, async (site, ct) =>
+        var sites = await siteService.GetAllSites();
+        var activeSites = request.Sites.Where(rs => sites.Any(s => s.Id == rs && s.isDeleted is false or null));
```

Comment:
Absolutely nothing wrong with this, but `.GetAllSites()` will run exactly this filter for you if `includeDeleted` is false, which it is by default. So you can skip it here if you like :)